crates/rolldown/src/ast_scanner/impl_visit.rs

```diff
@@ -184,7 +192,19 @@ impl<'me, 'ast> Visit<'ast> for AstScanner<'me> {
           if request.span().is_empty() {
             ImportRecordMeta::IS_UNSPANNED_IMPORT
           } else {
-            ImportRecordMeta::empty()
+            let mut meta = ImportRecordMeta::empty();
+            // traverse nearest ExpressionStatement and check if there are potential used
+            for ancestor in self.visit_path.iter().rev() {
```

Comment:
Should it be `self.visit_path[..self.vist_path.len].iter().rev()`? Bacause the path vec contains the node itself.

Maybe a `iter_ancestor` method?

Reply:
> Should it be `self.visit_path[..self.vist_path.len].iter().rev()`? Bacause the path vec contains the node itself.
> 
> Maybe a `iter_ancestor` method?

No, the `enter_node` called in `walk_call_expression`
```rs
    #[inline]
    pub fn walk_call_expression<'a, V: Visit<'a>>(visitor: &mut V, it: &CallExpression<'a>) {
        let kind = AstKind::CallExpression(visitor.alloc(it));
        visitor.enter_node(kind);
        visitor.visit_expression(&it.callee);
        if let Some(type_parameters) = &it.type_parameters {
            visitor.visit_ts_type_parameter_instantiation(type_parameters);
        }
        visitor.visit_arguments(&it.arguments);
        visitor.leave_node(kind);
    }
```

Reply:
Cool. Good to know this behavior